src/openeo_gfmap/features/feature_extractor.py

```diff
@@ -0,0 +1,285 @@
+"""Feature extractor functionalities. Such as a base class to assist the
+implementation of feature extractors of a UDF.
+"""
+
+from abc import ABC, abstractmethod
+
+import numpy as np
+import openeo
+import xarray as xr
+from openeo.udf import XarrayDataCube
+from openeo.udf.run_code import execute_local_udf
+from openeo.udf.udf_data import UdfData
+
+REQUIRED_IMPORTS = """
+import inspect
+from abc import ABC, abstractmethod
+
+import openeo
+from openeo.udf import XarrayDataCube, inspect
+from openeo.udf.run_code import execute_local_udf
+from openeo.udf.udf_data import UdfData
+
+import xarray as xr
+import numpy as np
+
+from typing import Union
+"""
+
+
+LAT_HARMONIZED_NAME = "GEO-LAT"
+LON_HARMONIZED_NAME = "GEO-LON"
+EPSG_HARMONIZED_NAME = "GEO-EPSG"
+
+# To fill in: EPSG_HARMONIZED_NAME, Is it pixel based and Feature Extractor class
+APPLY_DATACUBE_SOURCE_CODE = """
+LAT_HARMONIZED_NAME = "{}"
+LON_HARMONIZED_NAME = "{}"
```

Comment:
As mentioned below, I would recommend to use named placeholders instead of these error prone positional ones

Reply:
Done :)